### ISSUE TITLE
useUnit shape bug in old iOS Safari 

### ISSUE THREAD

**Issue by domosedov** [effector/effector#1020]
**What is the current behavior:**
Ошибка при использовании useUnit в форме массива

**Please provide the steps to reproduce and if possible a minimal demo of the problem via https://share.effector.dev, https://codesandbox.io or similar**

## Tested on iOS 15!

1) Clone repo
[Repo](https://github.com/domosedov/use-unit-old-safari-bug)

2) Install deps
`pnpm i`

3) Run dev mode
`pnpm dev`

4) Open in browser on `http://localhost:3000`

5) See [Counter.tsx](https://github.com/domosedov/use-unit-old-safari-bug/blob/main/src/app/counter.tsx)

**What is the expected behavior:**
Нет ошибок

**Which versions of effector packages, and which browser and OS are affected by this issue? Did this work in previous versions of effector?:**
"@effector/next": "^0.6.1",
"effector": "^23.0.0",
"effector-react": "^23.0.0",
"next": "14.0.3",

**Comment by zerobias** [comment]
iOS 15

**Comment by domosedov** [comment]
<img width="435" alt="Снимок экрана 2023-11-29 в 13 12 48" src="https://github.com/effector/effector/assets/42298502/64f86843-de84-4a7a-b45a-e20f8b171762">
<img width="344" alt="Снимок экрана 2023-11-29 в 13 13 00" src="https://github.com/effector/effector/assets/42298502/74fdc110-5f89-4d63-bc35-968a3bc2d73d">

**Comment by victordidenko** [comment]
Code
```js
var k
var a = [$count, incremented]
for (k in a) {
  console.log(k, a[k])
}
```
gives this
![telegram-cloud-photo-size-2-5422497005948752597-y](https://github.com/effector/effector/assets/15016227/66598d23-09f9-4c3e-b4a6-391ad2395d41)

Looks like some polyfill adds `.at` method to array, and absence of `.hasOwnProperty` check in `useUnit` causes this error.
https://github.com/effector/effector/blob/93abb4535df24894e6f975f42e77571cc0f5d201/src/react/apiBase.ts#L77

**Comment by AlexandrHoroshih** [comment]
Fixed in `effector-react@23.1.0`
